### PR TITLE
fixed port bug on LRP

### PR DESCRIPTION
**Pull request by gnovv**
Adds some test refactors and fixed a bug with the LRP Health Check when the CFProcess has no ports

### CODE REVIEW

File: controllers/controllers/workloads/cfprocess_controller.go [hunk 1]
```diff
@@ -103,6 +103,10 @@ func (r *CFProcessReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 			}
 		}
 
+		var lrpHealthCheckPort int32 = 0
```
Comment (Birdrock) [inline controllers/controllers/workloads/cfprocess_controller.go:106]: Isn't this redundant? The zero value for int32 is already 0 if I recall correctly.

File: controllers/controllers/workloads/cfprocess_controller.go [hunk 2]
```diff
@@ -103,6 +103,10 @@ func (r *CFProcessReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
+		if len(cfProcess.Spec.Ports) > 0 {
+			lrpHealthCheckPort = cfProcess.Spec.Ports[0]
```
Comment (Birdrock) [inline controllers/controllers/workloads/cfprocess_controller.go:108]: Out of scope for this, but I still feel it's pretty strange to just pick the first element as the health check. We don't even know what type of traffic whatever is listening on that port expects.

**Review by Birdrock — APPROVED** [review]
Isn't this redundant? The zero value of int32 is 0, iirc.

File: controllers/controllers/workloads/cfprocess_controller.go (hunk 1 above)
Comment (gnovv) [reply controllers/controllers/workloads/cfprocess_controller.go:106]: I guess I prefer to be explicit for now, we can change it later if people have concerns

**Pull request merged**